system/lpc176x/HardwarePWM.h

```diff
@@ -138,11 +134,9 @@ class HardwarePWM {
 
   // update the match register for a channel and set the latch to update on next period
   static inline void set_match(const pin_t pin, const uint32_t value) {
-    //work around for bug if MR1 == MR0
+    //work around for bug if MRn == MR0
     *match_register_ptr(pin) = value == LPC_PWM1->MR0 ? value + 1 : value;
-    // tried to work around latch issue by always setting all bits, was unsuccessful
-    LPC_PWM1->LER = util::bit_value(pin_has_pwm(pin));
```

Comment:
With no OR function at this point there would be inconsistent behaviour when more than one pwm was active.

Reply:
Indeed, other than re-enabling latching that's the only change in this PR, I wouldn't put it past myself to make such a stupid mistake and spend weeks trying to debug it (as I did at the time), but the comment seems to imply that I got frustrated at some point and just set all latch bits on every match update and it didn't work around the issue so I think not ORing the current register value was an artefact of that debugging and leaving latching disabled.

I'l do some more tests tomorrow, I'm mostly worried about the client application setting the match value but it not updating the signal when being spammed with match updates. (I think that was the original issue, .. I really have left this too long)

Reply:
Yes, been there too, with the bit function toolset this would be so easy to overlook. 
e.g. bitset vs bit value function template, one template applies OR and one does not.
+1 for more testing, quality = testing. Thanks for your work or this.